engine/execution/ingestion/engine.go

```diff
@@ -641,6 +645,10 @@ func (e *Engine) executeBlock(ctx context.Context, executableBlock *entity.Execu
 	if err != nil {
 		e.log.Err(err).Msg("failed in process block's children")
 	}
+
+	if e.executionDataPruner != nil {
```

Comment:
not really, but allowing it made updating the tests possible / easier. It would have been much more complicated to update the existing tests if it wasn't allowed to be nil.